crates/aptos-crypto/src/webauthn/webauthn_p256_sigs.rs

```diff
@@ -0,0 +1,238 @@
+    /// {R,n-S} as the canonical encoding of this signature to prevent malleability attacks. See
+    /// `check_s_malleability` for more detail
+    pub fn make_canonical(&self) -> WebAuthnP256Signature {
+        let signature = P256Signature::make_canonical(&self.p256_signature);
+        WebAuthnP256Signature {
+            actual_challenge: self.actual_challenge.clone(),
+            verification_data: self.verification_data.clone(),
+            p256_signature: signature,
+        }
+    }
+}
+
+/////////////////////
+// WebAuthn Traits //
+/////////////////////
+
+impl WebAuthnSignature for WebAuthnP256Signature {
+    type VerifyingKeyMaterial = WebAuthnP256PublicKey;
+    type SigningKeyMaterial = WebAuthnP256PrivateKey;
+
+    /// Generate a public key credential from public key credential bytes
+    fn try_from_public_key_credential_bytes(bytes: Vec<u8>) -> Result<Self, CryptoMaterialError> {
+        let credential: PublicKeyCredential = serde_json::from_slice(bytes.as_slice())
+            .map_err(|_e| CryptoMaterialError::DeserializationError)?;
+        let (v_data, signature, actual_challenge) =
+            parse_public_key_credential(&credential, COSEAlgorithm::ES256)
+                .map_err(|_e| CryptoMaterialError::DeserializationError)?;
+
+        let webauthn_p256_sig = WebAuthnP256Signature {
+            actual_challenge,
+            verification_data: v_data,
+            p256_signature: P256Signature(signature),
+        };
+
+        // Max size check of 200 bytes
+        if webauthn_p256_sig.to_bytes().len() > 200 {
+            return Err(CryptoMaterialError::DeserializationError);
+        }
+
+        Ok(webauthn_p256_sig)
+    }
+
+    /// Check to see if the provided arbitrary challenge matches with the
+    #[inline]
+    fn verify_arbitrary_challenge(
+        &self,
+        message: &[u8],
+        public_key: &Self::VerifyingKeyMaterial,
+    ) -> anyhow::Result<()> {
+        let is_equal = self.verify_expected_challenge_from_message_matches_actual(
+            message,
+            self.actual_challenge.as_ref(),
+        );
+
+        // Check if expected challenge and actual challenge match. If there's no match, throw error
+        if !is_equal {
+            return Err(anyhow!(
+                "Error: WebAuthn expected challenge did not match actual challenge"
+            ));
+        }
+
+        // Verify signature against verification data
+        public_key
+            .0
+             .0
+            .verify(self.verification_data.as_slice(), &self.p256_signature.0)
+            .map_err(|e| anyhow!("{}", e))
+            .and(Ok(()))
+    }
+}
+
+//////////////////////
+// Signature Traits //
+//////////////////////
+
+impl Signature for WebAuthnP256Signature {
+    type VerifyingKeyMaterial = WebAuthnP256PublicKey;
+    type SigningKeyMaterial = WebAuthnP256PrivateKey;
+
+    /// Verifies that the provided signature is valid for the provided message, going beyond the
```

Comment:
Would specify this is a feature of the underlying P256 implementation rather than the WebAuthP256 implementation

Reply:
Added a note